File: solutions/lua/triangle/1/triangle.lua
local triangle = {}

function triangle.kind(a, b, c)
    if a<=0 or b<=0 or c<=0 
    then
        return error('Input Error')
    end 
    if a+b<c or a+c<b or b+c<a 
    then
        return error('Input Error')
    end 
    if a==b and b==c 
    then
        return 'equilateral'
    end 
    if a==b or a==c or b==c
    then
        return 'isosceles'
    end 
    return 'scalene'
end

return triangle

File: solutions/cpp/triangle/1/triangle.cpp
#include "triangle.h"

namespace triangle {
    enum flavor kind(float s1, float s2, float s3){
        if (s1<=0 || s2<=0 || s3<=0) throw std::domain_error("zero side");
        if (s1>s2+s3 || s2>s1+s3 || s3>s1+s2) throw std::domain_error("zero side");
        if (s1==s2 && s1==s3) return flavor::equilateral;
        if (s1==s2 || s1==s3 || s2==s3) return flavor::isosceles;
        return flavor::scalene;
    }
}  // namespace triangle
